guide/src/conversions.md

```diff
@@ -2,27 +2,45 @@
 
 PyO3 provides some handy traits to convert between Python types and Rust types.
 
-## `.extract()`
+## `.extract()` and the `FromPyObject` trait
 
-The easiest way to convert a Python object to a Rust value is using `.extract()?`.
+The easiest way to convert a Python object to a Rust value is using
+`.extract()`.  It returns a `PyResult` with a type error if the conversion
+fails, so usually you will use something like
 
-## `ToPyObject` trait
+```
+let v: Vec<i32> = obj.extract()?;
+```
+
+This method is available for many Python object types, and can produce a wide
+variety of Rust types, which you can check out in the implementor list of
+[`FromPyObject`][FromPyObject].
+
+`FromPyObject` is also implemented for your own Rust types wrapped as Python
+objects (see [the chapter about classes](class.md)).  There, in order to both be
+able to operate on mutable references *and* satisfy Rust's rules of non-aliasing
+mutable references, you have to extract the PyO3 reference wrappers `PyRef<T>`
+and `PyRefMut<T>`.  They work like the reference wrappers of
+`std::cell::RefCell` and ensure (at runtime) that Rust borrows are allowed.
+
+
+## The `ToPyObject` trait
 
-[`ToPyObject`] trait is a conversion trait that allows various objects to be
+[`ToPyObject`] is a conversion trait that allows various objects to be
 converted into [`PyObject`][PyObject]. `IntoPy<PyObject>` serves the
 same purpose, except that it consumes `self`.
 
-## `FromPyObject` and `RefFromPyObject` trait
 
-## `*args` and `**kwargs` for python object call
+## `*args` and `**kwargs` for Python object calls
 
 There are several ways how to pass positional and keyword arguments to a Python object call.
```

Comment:
Should we maybe call these "Python function calls" instead of "Python object calls"?

Reply:
For me both are fine; there isn't a big distinction between calling functions and calling other callables, such as classes to create an instance.